### PR TITLE
DR-650 Adding Allure report to test pipeline

### PR DESCRIPTION
**Pull request by tamaragoldschmidt**
## Description

Adding the integration test framework allure report to the test pipeline

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by jope1 — APPROVED** [review]
Looks good to me

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+name: Run integration tests
+
+on:
+  workflow_call:
+    inputs:
+      github_environment:
```
Comment (timrickwood) [inline .github/workflows/integration-test.yaml:6]: Other than to echo out below is this used ? Typically we would use it to set env (eg to get secrets) ?

File: .github/workflows/pipeline-deploy-app-test.yaml [hunk 2]
```diff
@@ -111,35 +111,16 @@ jobs:
...
+      needs:
+        [
+          deploy-application-infrastructure,
+          build-and-deploy-application-code,
+          derive-deployment-workspace,
+        ]
+      name: Run integration tests
+      uses: ./.github/workflows/integration-test.yaml
+      with:
+        github_environment: test
```
Comment (timrickwood) [inline .github/workflows/pipeline-deploy-app-test.yaml:123]: See above ... is this used in callled workflow ?

File: .github/workflows/integration-test.yaml (hunk 1 above)
Comment (tamaragoldschmidt) [reply .github/workflows/integration-test.yaml:6]: No it is not used and neither is git_reference. Will remove both

File: .github/workflows/pipeline-deploy-app-test.yaml (hunk 2 above)
Comment (tamaragoldschmidt) [reply .github/workflows/pipeline-deploy-app-test.yaml:123]: Isn't this required to by action-stack?

**Review by timrickwood — APPROVED** [review]

**Review by jope1 — APPROVED** [review]

**Pull request merged**